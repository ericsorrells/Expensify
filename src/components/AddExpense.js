import React from 'react';
import ExpenseForm from './ExpenseForm';

const AddExpense = () => (
  <div>
    <h1>ADD EXPENSE</h1>
    <ExpenseForm />
  </div>
);

export default AddExpense;