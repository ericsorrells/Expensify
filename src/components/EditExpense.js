import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { editExpense, startRemoveExpense } from '../actions/expenses';

class EditExpense extends React.Component {
  onSubmit = (expense) => {
      this.props.editExpense(this.props.expense.id, expense);
      this.props.history.push('/');
  }
  
  onClick = () => {
    this.props.startRemoveExpense(this.props.expense);
    this.props.history.push('/');
  }

  render() {
    return(
      <div>
        <ExpenseForm
          expense={this.props.expense}
          onSubmit={this.onSubmit}
        />
        <button onClick={this.onClick }>
          Remove
        </button>
       </div>
    )
  }
}

const mapDispatchToProps = (dispatch, props) => ({
  editExpense: (id, expense) => dispatch(editExpense(id, expense)),
  startRemoveExpense: (expense) => dispatch(startRemoveExpense(expense)) 
});

const mapStateToProps = (state, props) => {
  return {
    expense: state.expenses.find((expense) => expense.id === props.match.params.id)
  }
}

export { EditExpense };
export default connect(mapStateToProps, mapDispatchToProps)(EditExpense);