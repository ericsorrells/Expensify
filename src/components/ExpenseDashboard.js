import React from 'react';

const ExpenseDashboard = () => (
  <div>Home PAGE!!!</div>
);

export default ExpenseDashboard;