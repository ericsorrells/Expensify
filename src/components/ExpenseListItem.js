import React from 'react';
import { removeExpense } from '../actions/expenses'
import { connect } from 'react-redux';

const ExpenseListItem = ({ dispatch, id, description, amount, createdAt }) => ( 
  <div>
    <h3>Description: {description}</h3>
    <p>Amount: {amount} |
    Created: {createdAt}</p>
    <button onClick={() => {
      dispatch(removeExpense({ id }))
    }}
    >
      Remove Me
    </button>
  </div>
)

export default connect()(ExpenseListItem)