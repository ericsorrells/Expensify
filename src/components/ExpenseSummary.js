import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import numeral from 'numeral';
import getExpenseTotal from '../selectors/expenses-total';

const ExpenseSummary = (props) => {
  const expenseWord = props.expenses.length === 1 ? 'expense' : 'expenses';
  const expenseTotal = numeral(getExpenseTotal(props.expenses) / 100).format('$0,0.00');
  return (
    <div>
      <h3>Viewing {props.expenses.length} {expenseWord} totaling {expenseTotal}</h3>
    </div>
  )
}

const mapStateToProps = (state) => {
  return {
    expenses: selectExpenses(state.expenses, state.filters) 
  }
} 

export { ExpenseSummary };
export default connect(mapStateToProps)(ExpenseSummary);