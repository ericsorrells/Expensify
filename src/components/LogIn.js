import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

const LogIn = ({ startLogin }) => (
  <div>
    <button onClick={ startLogin }>Log In!</button>
  </div>
)

const mapDispatchToProps = (dispatch) => ({
  startLogin: () => dispatch(startLogin())
});

export { LogIn };
export default connect(undefined, mapDispatchToProps)(LogIn);