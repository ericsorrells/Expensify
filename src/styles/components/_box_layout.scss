.box-layout {
  background: url('/images/bg.jpg');
  height: 100vh;  // 100% of viewport height
  width: 100vw; // 100% of viewport width
  background-size: cover; // should cover the entire area (?)
  display: flex;
  justify-content: center;
  align-items: center;
}

.box-layout__box {
  background: fade-out($color: white, $amount: 0.25);
  border-radius: 3px;
  text-align: center;
  width: 25rem;
  padding: $l-size $m-size;
}

.box-layout__title {
  margin: 0 0 $m-size 0;
  line-height: 1;
}