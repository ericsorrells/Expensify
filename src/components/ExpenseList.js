import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses'

const ExpenseList = (props) => (
  <div>
    <h1>Expense List!!!</h1>
    {
      props.expenses.length === 0 ?
        (<p>No Expenses</p>) :
        (
          <div>
            {props.expenses.map((expense, index) =>
              <ExpenseListItem {...expense} key={expense.id} />
            )}
          </div>
        )
    }
  </div>
);

const mapStateToProps = (state) => {
  return {
    expenses: selectExpenses(state.expenses, state.filters) 
  }
} 

export { ExpenseList };
export default connect(mapStateToProps)(ExpenseList);
