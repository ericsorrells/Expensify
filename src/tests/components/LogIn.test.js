import React from 'react';
import { shallow } from 'enzyme';
import { LogIn } from '../../components/LogIn';

test('<LogIn />: renders correctly', () => {
  const component = shallow(<LogIn startLogin="() => {}"/>);
  expect(component).toMatchSnapshot();
});

test('should call startLogIn on button click', () => {
  const startLogin = jest.fn();
  const component = shallow(<LogIn startLogin={startLogin} />);
  component.find('button').simulate('click')
  expect(startLogin).toHaveBeenCalled(); 
})


