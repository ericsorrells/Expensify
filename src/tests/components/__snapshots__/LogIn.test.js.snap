// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogIn />: renders correctly 1`] = `
<div>
  <button
    onClick="() => {}"
  >
    Log In!
  </button>
</div>
`;
